docs/guides/guide-vmanomaly-vmalert.md

```diff
@@ -0,0 +1,403 @@
+node_cpu_seconds_total{cpu="0",mode="user"} 12378.05
+node_cpu_seconds_total{cpu="1",mode="idle"} 94386.53
+node_cpu_seconds_total{cpu="1",mode="iowait"} 51.22
+node_cpu_seconds_total{cpu="1",mode="irq"} 0
+node_cpu_seconds_total{cpu="1",mode="nice"} 0
+node_cpu_seconds_total{cpu="1",mode="softirq"} 39.18
+node_cpu_seconds_total{cpu="1",mode="steal"} 0
+node_cpu_seconds_total{cpu="1",mode="system"} 1378.79
+node_cpu_seconds_total{cpu="1",mode="user"} 14241.59
+node_cpu_seconds_total{cpu="2",mode="idle"} 93929.1
+node_cpu_seconds_total{cpu="2",mode="iowait"} 50.93
+node_cpu_seconds_total{cpu="2",mode="irq"} 0
+node_cpu_seconds_total{cpu="2",mode="nice"} 0
+node_cpu_seconds_total{cpu="2",mode="softirq"} 30.41
+node_cpu_seconds_total{cpu="2",mode="steal"} 0
+node_cpu_seconds_total{cpu="2",mode="system"} 1368.52
+node_cpu_seconds_total{cpu="2",mode="user"} 14722.17
+node_cpu_seconds_total{cpu="3",mode="idle"} 93554.49
+```
+Here, `node_cpu_seconds_total` metric tells us how many seconds each CPU spent doing each type of work: _user_, _system_, _iowait_, _idle_, _irq&softirq_, _guest_, or _steal_.
+These modes are mutually exclusive. A high _iowait_ means that you are disk or network bound, high user or system means that you are CPU bound.
+
+We'll send this node_exporter data into VictoriaMetrics later in this tutorial. But lets just note that `node_cpu_seconds_total` metric is a growing counter, so to calculate the per-second values we will use the rate function: `rate(node_cpu_seconds_total)`.
+Here is how this query may look like in Grafana:
+![node_cpu_rate_graph](guide-vmanomaly-node-cpu-rate-graph.png "node_cpu_rate_graph")
+
+This query result will generate 8 data vectors per each cpu, and we will use them as an input for our VM Anomaly Detection. VmAnomaly will start learning configured model type separately for each of the data vectors.
+
+
+______________________________
+
+## VmAnomaly configuration and parameter description
+Here is an example of the config file `vmanomaly_config.yml`.
+```
+scheduler:
+  class: "scheduler.periodic.PeriodicScheduler"
+  infer_every: "30s"
+  fit_window: "2d"
+
+model:
+  class: "model.prophet.ProphetModel"
+  interval_width: 0.8
+
+reader:
+  class: "reader.vm.VmReader"
+  datasource_url: "http://victoriametrics:8428/"
+  queries:
+    node_cpu_rate: "rate(node_cpu_seconds_total)"
+
+writer:
+  class: "writer.vm.VmWriter"
+  datasource_url: "http://victoriametrics:8428/"
+
+```
+**Parameter description**:
+There are 4 main sections in config file:
+
+`scheduler` - defines how often to run and make inferences, as well as what timerange to use to train the model. 
+
+`model` - specific model parameters and configurations, 
+
+`reader` - where and how to read data from,
+
+`writer` - where and how to write the generated output.
+
+Let's look into parameters in each section:
+
+* `scheduler` 
+    As we need to set an alert for it, we will use periodic "scheduler.periodic.PeriodicScheduler".
```

Comment:
Are there any other classes?

Reply:
There is one-off class for one-time testing and tuning the model.